client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -315,9 +318,17 @@ private Optional<String> getModelNamePrefix(final Config config, final Path open
     }
 
     private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, final Config config) {
-        return config.getOptionalValue(getGlobalConfigName(INPUT_BASE_DIR), String.class).map(inputBaseDir -> {
+        return getBaseDirRelativeToModule(config, getGlobalConfigName(INPUT_BASE_DIR), sourceDir);
+    }
+
+    private Optional<String> getTemplateBaseDirRelativeToModule(final Path sourceDir, final Config config) {
```

Comment:
The template base directory must be relative to `src/main/resources`.

Reply:
IIRC, `baseDir` is not meant to be used by the templates but by the open API folder on the `src/main`.

Reply:
I just generalized the method for getting a base dir for a certain config value for simplicity reasons.

Reply:
I am sorry, I guess i do not get your point 😓 
Isn't it unnecessary to check if the templates should be distributed? I thought templates are relevant for the code generation step.
The initial issue, why i created this PR i the first place, was that in my project we use gradle and resources are copied to `build/resources/main/templates` assuming they are stored in `src/main/resources/templates`. So i had to create a custom task copying the templates to `build/classes/templates` to be able to correctly generate code.

Reply:
My point is that the code should fetch the custom templates from `src/main`, not the project's root.

Reply:
I just updated the method. The sourceDir is taken and the default input base dir stripped to get the source root dir. Custom locations are now enforced to be under `src/main`.